x/profiles/keeper/keeper_relationships.go

```diff
@@ -0,0 +1,76 @@
+package keeper
+
+import (
+	"bytes"
+	"fmt"
+
+	sdk "github.com/cosmos/cosmos-sdk/types"
+	"github.com/desmos-labs/desmos/x/profiles/types"
+)
+
+// StoreRelationship allows to store the given receiver returning an error if he's already present.
+func (k Keeper) StoreRelationship(ctx sdk.Context, user, receiver sdk.AccAddress) error {
+	store := ctx.KVStore(k.StoreKey)
+	key := types.RelationshipsStoreKey(user)
+	var relationships []sdk.AccAddress
+	k.Cdc.MustUnmarshalBinaryBare(store.Get(key), &relationships)
+
+	for _, addr := range relationships {
+		if addr.Equals(receiver) {
+			return fmt.Errorf("relationship already exists with %s", receiver)
+		}
+	}
```

Comment:
Instead of performing this check here, I think we can: 

1. Extract these lines in a new method, maybe something like `DoesRelationshipExist`
2. Call the new method from the handler
3. Inside this method, if the relationship already exists then simply not add it to the array instead of throwing an error

Reply:
I don't think extracting this is the better choice since we use this only here and, with a method like `DoesRelationshipExist`, we'll need to access two times to the store instead of one inside the `msg` `Handler`.

Reply:
@bragaz Ok, I see. Then let's keep it that way